### PR TITLE
Add IP address (host) label to XRootD access metrics

### PR DESCRIPTION
**Pull request by patrickbrophy**
This PR addresses #1425. This PR adds a IP address (host) label to XRD access metrics (xrootd_transfer_readv_segments_count, xrootd_transfer_operations_count, and xrootd_transfer_bytes). This PR also updated some tests to handle the new label.

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -102,3 +103,49 @@ func CheckValidQuery(transferUrl *url.URL) (err error) {
...
+	mask := net.CIDRMask(64, 128)
+	maskedIP := ip.Mask(mask)
+	return maskedIP.String()
+}
+
+// MaskIP applies the appropriate subnet mask to the input IP address.
+func MaskIP(ipStr string) string {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		// Invalid IP
```
Comment (bbockelm) [inline utils/utils.go:135]: Does this cause an explosion in cardinality if a serves is using hostnames and not an IP?

Good habits would have an indication to the caller that a special case occurred. Maybe also return a Boolean?

File: utils/utils.go [hunk 2]
```diff
@@ -102,3 +103,49 @@ func CheckValidQuery(transferUrl *url.URL) (err error) {
...
+		return ipStr
+	}
+
+	if ip.To4() != nil {
+		// IPv4
+		return IPv4Mask(ipStr)
```
Comment (bbockelm) [inline utils/utils.go:141]: Can we avoid parsing the IP string twice?

**Review by bbockelm — CHANGES_REQUESTED** [review]
Three requests:
- can we leave out the unrelated movement of structs to a separate module?  That causes code churn unrelated to the PR (but may be fine for a separate PR).
- there's a Xrootd setting, `xrd.network nodnr`, that should disable the use of reverse DNS to create a hostname in the monitoring records.  Can we try setting that here?  See https://xrootd.slac.stanford.edu/doc/dev56/xrd_config.htm#_Toc152596878
- given we are using networks and not hostnames, let's rename the labels to "network" perhaps?  Looking around on Google, it's not clear if there's a standard practice here.

**Comment by patrickbrophy** [comment]
I'm closing this PR to separate out concerns. I'll open a PR for this metric and a separate one for moving around the structs from the utils folder.

**Pull request closed without merging**